bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,810 @@
+                let status = Status {
+                    fsm_state: self.fsm.state().name(),
+                    peers: self.peers.clone(),
+                    connections: self
+                        .established_connections
+                        .iter()
+                        .map(|(id, handle)| (id.clone(), handle.addr))
+                        .collect(),
+                    accepted_connections: self
+                        .accepted_connections
+                        .keys()
+                        .cloned()
+                        .collect(),
+                    negotiating_connections: self
+                        .initiating_connections
+                        .keys()
+                        .cloned()
+                        .collect(),
+                };
+                let _ = responder.send(status);
+            }
+            NodeApiRequest::Shutdown => {
+                self.shutdown = true;
+                // Shutdown all connection processing tasks
+                for (_, handle) in &self.accepted_connections {
+                    let _ = handle.tx.send(MainToConnMsg::Close).await;
+                }
+                for (_, handle) in &self.initiating_connections {
+                    let _ = handle.tx.send(MainToConnMsg::Close).await;
+                }
+                for (_, handle) in &self.established_connections {
+                    let _ = handle.tx.send(MainToConnMsg::Close).await;
+                }
+            }
+        }
+    }
+
+    // Route messages to their destination connections
+    async fn deliver_envelopes(&mut self) {
+        for envelope in self.fsm.drain_envelopes() {
+            if let Some(conn_handle) =
+                self.established_connections.get(&envelope.to)
+            {
+                debug!(
+                    self.log,
+                    "Sending {:?} to {}", envelope.msg, envelope.to
+                );
+                if let Err(e) = conn_handle
+                    .tx
+                    .send(MainToConnMsg::Msg(Msg::Fsm(envelope.msg)))
+                    .await
+                {
+                    warn!(self.log, "Failed to send {e:?}");
+                }
+            } else {
+                warn!(self.log, "Missing connection to {}", envelope.to);
+            }
+        }
+    }
+
+    // Perform any operations required by a given `ApiOutput`, such as
+    // persisting state, and then inform any callers (via outstanding responders)
+    // of the result.
+    async fn handle_api_output(&mut self, output: ApiOutput) {
+        info!(self.log, "Fsm output = {output:?}");
+        match output {
+            // Initialization is mutually exclusive
+            ApiOutput::PeerInitialized | ApiOutput::RackInitComplete => {
+                if let Some(responder) = self.init_responder.take() {
+                    let _ = responder.send(Ok(()));
+                }
+                // TODO: Persistence
```

Comment:
What does "persistence" mean here?

Reply:
It means saving the state of the Fsm to disk so the peer can be restarted. It's coming in a follow up PR.